woocommerce/google-listings-and-ads
# [GTIN] GTIN Migration API Controller

### Changes proposed in this Pull Request:



Closes part of https://github.com/woocommerce/google-listings-and-ads/issues/2616 as part of pcTzPl-2p2-p2.

_Replace this with a good description of your changes & reasoning._


### Screenshots:

<img width="1295" alt="Screenshot 2024-10-29 at 19 23 48" src="https://github.com/user-attachments/assets/8fb6557c-548e-422a-bbd4-8727903e4eff">



### Detailed test instructions:


1. Request `POST /wp-json/wc/gla/gtin-migration-start` 
2. See `{"status":"success","message":"GTIN Migration successfully started."}`
3. Repeat step 1
4. See `{"status":"error","message":"GTIN Migration cannot be scheduled."}`
5. Complete the job (all the batches)
6. Go to options and see `gla_gtin_migration_started` as 1 - true.


### Additional details:

- Tests coming in other PR
- Nonce protection coming in other PR


### Changelog entry

> Add - GTIN Migration API Controller
